### PR TITLE
[13.0][MIG] base_export_manager: Migration to 13.0

### PR DESCRIPTION
**Pull request by Tardo**
cc @tecnativa TT21616

### CODE REVIEW

**Review by manuelcalerosolis — APPROVED** [review]
Functional tested ok!

File: base_export_manager/README.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fserver--ux-lightgray.png?logo=github
+    :target: https://github.com/OCA/server-ux/tree/12.0/base_export_manager
+    :alt: OCA/server-ux
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_export_manager
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/250/12.0
```
Comment (sergio-teruel) [inline base_export_manager/README.rst]: Re-generate file

File: base_export_manager/models/ir_exports.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class IrExports(models.Model):
+    _inherit = "ir.exports"
+
+    name = fields.Char(required=True)
+    resource = fields.Char(
+        required=False, readonly=True, help="Model's technical name."
```
Comment (sergio-teruel) [inline base_export_manager/models/ir_exports.py]: Why require is False, it's already False

File: base_export_manager/models/ir_exports.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            s.model_id = self._get_model_id(s.resource)
+
+    @api.onchange("model_id")
+    def _inverse_model_id(self):
+        """Get the resource from the model."""
+        for s in self:
+            s.resource = s.model_id.model
+
+    @api.onchange("resource")
+    def _onchange_resource(self):
```
Comment (sergio-teruel) [inline base_export_manager/models/ir_exports.py:24]: In v13.0 I think you can remove onchange method for views, compute method will behave as an onchange...

File: base_export_manager/models/ir_exports.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        """Void fields if model is changed in a view."""
+        for s in self:
+            s.export_fields = False
+
+    @api.model
+    def _get_model_id(self, resource):
```
Comment (sergio-teruel) [inline base_export_manager/models/ir_exports.py]: To consistent, this method return a record instead an id... so I think mybe the name could be _get_model

File: base_export_manager/models/ir_exports.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        help="Database model to export.",
+    )
+
+    @api.depends("resource")
+    def _compute_model_id(self):
+        """Get the model from the resource."""
+        for s in self:
+            s.model_id = self._get_model_id(s.resource)
+
+    @api.onchange("model_id")
```
Comment (sergio-teruel) [inline base_export_manager/models/ir_exports.py]: I think you can change for a depends

File: base_export_manager/models/ir_exports.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    def _get_model_id(self, resource):
+        """Return a model object from its technical name.
+
+        :param str resource:
+            Technical name of the model, like ``ir.model``.
+        """
+        return self.env["ir.model"].search([("model", "=", resource)])
+
+    @api.model_create_multi
+    def create(self, vals_list):
```
Comment (sergio-teruel) [inline base_export_manager/models/ir_exports.py:42]: Why create_multi decorator if after you call to super with the last vals_list item??

File: base_export_manager/models/ir_exports_line.py [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+
+from odoo import _, api, exceptions, fields, models
+
+
+class IrExportsLine(models.Model):
+    _inherit = "ir.exports.line"
+    _order = "sequence,id"
+
+    name = fields.Char(
+        required=False,
```
Comment (sergio-teruel) [inline base_export_manager/models/ir_exports_line.py]: Remove required=False...

File: base_export_manager/models/ir_exports_line.py [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+        readonly=True,
```
Comment (sergio-teruel) [inline base_export_manager/models/ir_exports_line.py]: If you set readonly, the inverse method does not write... so chage to False or remove it

File: base_export_manager/models/ir_exports_line.py [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+                lines = one.search(
+                    [("export_id", "=", one.export_id.id), ("name", "=", one.name)]
+                )
+                if len(lines) > 1:
+                    raise exceptions.ValidationError(
+                        _("Field '%s' already exists") % one.name
+                    )
+
+    @api.onchange("name")
+    def _onchange_name(self):
```
Comment (sergio-teruel) [inline base_export_manager/models/ir_exports_line.py]: I think you can remove this method and add api.depends("name") to _inverse_name method

File: base_export_manager/static/src/js/base_export_manager.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+            const render_export_enalble = jQuery.inArray(
+                this.env.model, session.export_models);
+            if (!session.is_superuser && sectionCode === 'other' &&
+                    items.length && render_export_enalble < 0) {
+                _items = _.reject(_items, {label:_t("Export")});
+            }
+            this._super(sectionCode, _items);
+        },
+    });
+});
```
Comment (sergio-teruel) [inline base_export_manager/static/src/js/base_export_manager.js:36]: This file does not show technical name...
V12.0
Selección_588
V13.0
Selección_589
See https://github.com/odoo/odoo/blob/43a115e94bea04a0533e57fb3bc1e52571a74c0b/addons/web/static/src/js/widgets/data_export.js#L76

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

**Comment by Tardo** [comment]
OK, changes done! thanks @sergio-teruel

**Comment by sergio-teruel** [comment]
@Tardo Sorry I tested and you need the on_change ....

**Comment by pedrobaeza** [comment]
Are you sure? What about adding a `depends`?

**Comment by sergio-teruel** [comment]
I'm sure...

**Comment by pedrobaeza** [comment]
Well, onchanges are going to disappear, so if you say that, then we won't have any way to implement that "missing feature"?

**Comment by Tardo** [comment]
OK! yes... inverse methods appears not be affected by decorators as "depends". OTOH these onchange is really never used.

thx again @sergio-teruel

**Review by sergio-teruel — APPROVED** [review]

File: base_export_manager/models/ir_model_access.py [hunk 11]
```diff
@@ -8,4 +8,6 @@
 class IrModelAccess(models.Model):
     _inherit = "ir.model.access"
 
+    # WARN: This can't be used in 'check()'
+    # See 'base > ir_model > line 1530
```
Comment (yajo) [inline base_export_manager/models/ir_model_access.py]: :stop_sign: This comment is tricky. If a new commit enters that file, the line number could change.

It's better if you provide an absolute link to the code, from github.

File: base_export_manager/models/res_users.py [hunk 12]
```diff
@@ -11,6 +11,9 @@ class ResUsers(models.Model):
 
     @api.model
     def fetch_export_models(self):
+        """
+        Gets all models where the user has export access
+        """
```
Comment (yajo) [inline base_export_manager/models/res_users.py]:
```suggestion
        """Gets all models where the user has export access"""
```

See https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

File: base_export_manager/static/src/xml/base.xml [hunk 13]
```diff
@@ -2,8 +2,8 @@
 <templates id="template" xml:space="preserve">
 
     <t t-extend="Export.TreeItems">
-        <t t-jquery=".o_tree_column" t-operation="append">
-             (<t t-esc="field.id"/>)
+        <t t-jquery="div[t-attf-class='o_tree_column']" t-operation="append">
```
Comment (yajo) [inline base_export_manager/static/src/xml/base.xml]:
```suggestion
        <t t-jquery="div[t-attf-class~='o_tree_column']" t-operation="append">
```

See https://api.jquery.com/attribute-contains-word-selector/

**Review by yajo — CHANGES_REQUESTED** [review]
One stop point and a couple of little suggestions.

Minor details, it looks very good overall.

Also rebase please, to fix those conflicts with .isort.cfg

**Comment by Tardo** [comment]
Changes done!

File: base_export_manager/README.rst [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+You can also go to Settings > Technical > Security > Access Rights.
+
+Usage
+=====
+
+You can create the export profiles as you are used to:
+
+* Go to any list view.
+* Check some records.
+* Press *More > Export*.
```
Comment (chienandalu) [inline base_export_manager/README.rst]:
```suggestion
* CLick *Action > Export*.
```

File: base_export_manager/README.rst [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+* Use the wizard to choose the columns to export.
+* Press *Save fields list*.
```
Comment (chienandalu) [inline base_export_manager/README.rst]:
```suggestion
* Click on the *Template* dropdown and select *New template*
```

File: base_export_manager/README.rst [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+* Give it a name.
+* Press *OK*.
```
Comment (chienandalu) [inline base_export_manager/README.rst]:
```suggestion
* Click on the floppy disk icon to save it.
```

**Review by chienandalu — APPROVED** [review]
Tested on runbot. Some minor remarks

**Comment by Tardo** [comment]
Changes done

**Comment by pedrobaeza** [comment]
@Yajo please update your review

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
Rebase to fix conflicts please.

File: .isort.cfg [hunk 17]
```diff
@@ -10,3 +10,4 @@ known_odoo=odoo
 known_odoo_addons=odoo.addons
 sections=FUTURE,STDLIB,THIRDPARTY,ODOO,ODOO_ADDONS,FIRSTPARTY,LOCALFOLDER
 default_section=THIRDPARTY
+known_third_party = dateutil,lxml,mock,psycopg2,setuptools
```
Comment (pedrobaeza) [inline .isort.cfg:13]: This should be restored to previous state

Comment (Tardo) [reply .isort.cfg:13]: pre-commit all times adds this line :/

Comment (pedrobaeza) [reply .isort.cfg:13]: If you haven't rebased, probably. And make sure you have latest pre-commit installed in your repo (`pre-commit install -f`)

Comment (yajo) [reply .isort.cfg:13]: Don't blame him :stuck_out_tongue_winking_eye:. pre-commit config is outdated in this repo:

https://github.com/OCA/server-ux/blob/332cc53946135e9da153ba1afd7f99a4fd65fa1d/.pre-commit-config.yaml#L54-L57

Comment (pedrobaeza) [reply .isort.cfg:13]: How can that be?

Comment (yajo) [reply .isort.cfg:13]: @sbidoul did the mass update, isn't it? Did you forget this repo @sbidoul? :thinking:

Comment (sbidoul) [reply .isort.cfg:13]: @yajo I did not forget, but apparently the update failed and I did not notice it. See #155

**Pull request merged**